src/CompletionProvider.php

```diff
@@ -134,30 +132,25 @@ public function provideCompletion(PhpDocument $doc, Position $pos): CompletionLi
             || $node instanceof Node\Expr\StaticPropertyFetch
             || $node instanceof Node\Expr\ClassConstFetch
         ) {
-            if (!is_string($node->name)) {
-                // If the name is an Error node, just filter by the class
+            // If the name is an Error node, just filter by the class
+            if ($node instanceof Node\Expr\MethodCall || $node instanceof Node\Expr\PropertyFetch) {
+                // For instances, resolve the variable type
+                $prefixes = DefinitionResolver::getFqnsFromType(
+                    $this->definitionResolver->resolveExpressionNodeToType($node->var)
+                );
+            } else {
+                $prefixes = [$node->class instanceof \PhpParser\Node\Name ? (string)$node->class : ''];
```

Comment:
For some reason previous condition returns false for case like 'TestClass::$st'. Most probably because there is namespace \PhpParser\Node\Name and there is class 'Name' in namespace `\PhpParser\Node`. I change it and I wanted to ask what will be the best solution for this :) You were faster.